Software/Python/easygopigo.py

```diff
@@ -0,0 +1,405 @@
+
+
+ANALOG = 1
+DIGITAL = 0
+SERIAL = -1
+I2C = -2
+
+##########################
+
+
+class Sensor():
+    '''
+    Base class for all sensors
+    Class Attributes:
+        port : string - user-readable port identification
+        portID : integer - actual port id
+        pinmode : "INPUT" or "OUTPUT"
+        pin : 1 for ANALOG, 0 for DIGITAL
+        descriptor = string to describe the sensor for printing purposes
+    Class methods:
+        setPort / getPort
+        setPinMode / getPinMode
+        isAnalog
+        isDigital
+    '''
+    def __init__(self, port, pinmode):
+        '''
+        port = one of PORTS keys
+        pinmode = "INPUT", "OUTPUT", "SERIAL" (which gets ignored)
+        '''
+        debug("Sensor init")
+        debug(pinmode)
+        self.setPort(port)
+        self.setPinMode(pinmode)
+        if pinmode == "INPUT" or pinmode == "OUTPUT":
+            gopigo.pinMode(self.getPortID(), self.getPinMode())
+
+    def __str__(self):
+        return ("{} on port {}".format(self.descriptor, self.getPort()))
+
+    def setPort(self, port):
+        self.port = port
+        self.portID = PORTS[self.port]
+
+    def getPort(self):
+        return (self.port)
+
+    def getPortID(self):
+        return (self.portID)
+
+    def setPinMode(self, pinmode):
+        self.pinmode = pinmode
+
+    def getPinMode(self):
+        return (self.pinmode)
+
+    def isAnalog(self):
+        return (self.pin == ANALOG)
+
+    def isDigital(self):
+        return (self.pin == DIGITAL)
+
+    def set_descriptor(self, descriptor):
+        self.descriptor = descriptor
+##########################
+
+
+class DigitalSensor(Sensor):
+    '''
+    Implements read and write methods
+    '''
+    def __init__(self, port, pinmode):
+        debug("DigitalSensor init")
+        self.pin = DIGITAL
+        Sensor.__init__(self, port, pinmode)
+
+    def read(self):
+        '''
+        tries to get a value up to 10 times.
+        As soon as a valid value is read, it returns either 0 or 1
+        returns -1 after 10 unsuccessful tries
+        '''
+        okay = False
+        error_count = 0
+        while not okay and error_count < 10:
+            try:
+                rtn = int(gopigo.digitalRead(self.getPortID()))
+                okay = True
+            except:
+                error_count += 1
+        if error_count > 10:
+            return -1
+        else:
+            return rtn
+
+    def write(self, power):
+        self.value = power
+        return gopigo.digitalWrite(self.getPortID(), power)
+##########################
+
+
+class AnalogSensor(Sensor):
+    '''
+    implements read and write methods
+    '''
+    def __init__(self, port, pinmode):
+        debug("AnalogSensor init")
+        self.value = 0
+        self.pin = ANALOG
+        Sensor.__init__(self, port, pinmode)
+
+    def read(self):
+        self.value = gopigo.analogRead(self.getPortID())
+        return self.value
+
+    def write(self, power):
```

Comment:
Not sure I understand your comment. I'm simply passing it on to 
https://github.com/DexterInd/GoPiGo/blob/master/Software/Python/gopigo.py#L344